src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/InterfaceMemberImpliedModifierCheck.java

```diff
@@ -0,0 +1,423 @@
+ *   public abstract Address createAddress(String addressLine, String city);
+ *
+ *   List&lt;Address&gt; findAddresses(String city);  // violation
+ *
+ *   // check enforces default methods are explicitly declared "public"
+ *   public default Address createAddress(String city) {
+ *     return createAddress(UNKNOWN, city);
+ *   }
+ *
+ *   default Address createOtherAddress() {  // violation
+ *     return createAddress(OTHER, OTHER);
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * Rationale for these checks: Methods, fields and nested types are treated differently
+ * depending on whether they are part of an interface or part of a class. For example, by
+ * default methods are package-scoped on classes, but public on interfaces. However, from
+ * Java 8 onwards, interfaces have changed to be much more like abstract classes.
+ * Interfaces now have static and instance methods with code. Developers should not have to
+ * remember which modifiers are required and which are implicit. This check allows the simpler
+ * alternative approach to be adopted where the implied modifiers must always be coded explicitly.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code violateImpliedPublicField} - Control whether to enforce that {@code public}
+ * is explicitly coded and not implicitly added by the compiler on interface fields (true).
+ * Default value is {@code true}.
+ * </li>
+ * <li>
+ * Property {@code violateImpliedStaticField} - Control whether to enforce that {@code static}
+ * is explicitly coded and not implicitly added by the compiler on interface fields (true).
+ * Default value is {@code true}.
+ * </li>
+ * <li>
+ * Property {@code violateImpliedFinalField} - Control whether to enforce that {@code final}
+ * is explicitly coded and not implicitly added by the compiler on interface fields (true).
+ * Default value is {@code true}.
+ * </li>
+ * <li>
+ * Property {@code violateImpliedPublicMethod} - Control whether to enforce that {@code public}
+ * is explicitly coded and not implicitly added by the compiler on interface methods (true).
+ * Default value is {@code true}.
+ * </li>
+ * <li>
+ * Property {@code violateImpliedAbstractMethod} - Control whether to enforce that {@code abstract}
+ * is explicitly coded and not implicitly added by the compiler on interface methods (true).
+ * Default value is {@code true}.
+ * </li>
+ * <li>
+ * Property {@code violateImpliedPublicNested} - Control whether to enforce that {@code public}
+ * is explicitly coded and not implicitly added by the compiler on interface nested types (true).
+ * Default value is {@code true}.
+ * </li>
+ * <li>
+ * Property {@code violateImpliedStaticNested} - Control whether to enforce that {@code static}
+ * is explicitly coded and not implicitly added by the compiler on interface nested types (true).
+ * Default value is {@code true}.
+ * </li>
+ * </ul>
+ * <p>
+ * This example checks that methods, fields and nested types are explicitly specified
```

Comment:
> checks that methods ...

checks that all implicit modifiers on methods ...

Reply:
Done